src/libs/ReportUtils.ts

```diff
@@ -1609,6 +1609,33 @@ function getRoomWelcomeMessage(report: OnyxEntry<Report>): WelcomeMessage {
     return welcomeMessage;
 }
 
+function getWelcomeMessage(report: OnyxEntry<Report>): WelcomeMessage {
+    const welcomeMessage: WelcomeMessage = {showReportName: true};
+    if (isChatRoom(report)) {
+        return getRoomWelcomeMessage(report);
+    }
+
+    if (isPolicyExpenseChat(report)) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartOne');
+        welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartTwo');
+        welcomeMessage.phrase3 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartThree');
+        return welcomeMessage;
+    }
+
+    if (isSelfDM(report)) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistorySelfDM');
+        return welcomeMessage;
+    }
+
+    if (isSystemChat(report)) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistorySystemDM');
+        return welcomeMessage;
+    }
+
+    welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistory');
+    return welcomeMessage;
```

Comment:
Let's wrap this with `isDefault` condition for consistency with the rest.

Reply:
Currently we have the same logic in three places:
- ReportWelcomeText
- getWelcomeMessage
- getReportBeginningOfChatHistoryMessage

Can we simplify this and avoid the repeated logic?

Reply:
> Let's wrap this with isDefault condition for consistency with the rest.

I updated.

Reply:
> Can we simplify this and avoid the repeated logic?

I think it's hard because for `ReportWelcomeText`, each case we render with different UI and for `getReportBeginningOfChatHistoryMessage` we need add other texts for each case to concat the message.